### DISCUSSION TITLE
Legal use of software - Self-Hosted Azure DevOps agents- Visual Studio Enterprise

### DISCUSSION THREAD

**Discussion by gonchalo620** [actions/virtual-environments#3271] [Installed software]
We want to use Packer images to deploy Self-Hosted Agents in Azure DevOps with Scale Set strategy, for the continuous integration and deployment process at the enterprise level.
We have a concern, we can use all the software that are configured in the configuration files of the packer images https://github.com/actions/virtual-environments/tree/main/images, to build and deploy applications within the company without incurring illegal actions? For example, visual studio enterprise https://github.com/actions/virtual-environments/blob/main/images/win/scripts/Installers/Install-VS.ps1.

There is a similar question, but the answer is not accurate. https://github.community/t/github-hosted-visual-studio-enterprise-licensing/120487

Extra information:
https://github.com/actions/virtual-environments/blob/main/docs/software-and-images-guidelines.md
https://github.com/actions/virtual-environments/blob/main/LICENSE

**Comment by AlenaSviridenko** [comment]
Hi @gonchalo620,
yes, it is legal to use our templates to build a self-hosted agent. We install all software from public sources.

**Reply by gonchalo620** [reply]
Thank you very much.
